module/zfs/zfs_vfsops.c

```diff
@@ -1778,8 +1778,16 @@ zfsvfs_teardown(zfsvfs_t *zfsvfs, boolean_t unmounting)
 	 * Evict cached data. We must write out any dirty data before
 	 * disowning the dataset.
 	 */
-	if (!zfs_is_readonly(zfsvfs))
+	objset_t *os = zfsvfs->z_os;
+	boolean_t os_dirty = B_FALSE;
+	for (int t = 0; t < TXG_SIZE; t++) {
+		os_dirty = dmu_objset_is_dirty(os, t);
+		if (os_dirty)
+			break;
```

Comment:
[nit] I think this would be slightly easier to read as:
```
if (dmu_objset_is_dirty(...)) {
  os_dirty = B_TRUE;
  break;
}
```

Reply:
Thanks Matt I made that change.